src/content/plugins/split-chunks-plugin.md

```diff
@@ -0,0 +1,220 @@
+* Condition 2: `helpers` is bigger than 30kb
+* Condition 3: Number of parallel requests at the import calls is 2
+* Condition 4: Doesn't affect request at initial page load
+
+Putting the content of `helpers` into each chunk will result into its code being downloaded twice. By using a separate chunk this will only happen once. We pay the cost of an additional request, which could be considered a tradeoff. That's why there is a minimum size of 30kb.
+
+T> With the new `optimizations.splitChunks.chunks: "all"` option the same would happend for initial chunks. Chunks can even be shared between entrypoints and on-demand loading.
+
+## Configuration
+
+For these people that like to have more control over this functionality, webpack provides a set of options to better fit your needs.
+
+W> If you are manually changing the split configuration, measure the impact of the changes to see and make sure there's a real benefit. The defaults are choosen to fit best practices of web performance.
+
+### Configuring cache groups
+
+The defaults assigns all modules from `node_modules` to a cache group called `vendors` and all modules duplicated in at least 2 chunks to a change group `default`.
+
+A module can be assigned to multiple cache groups. The optimization then prefers the cache group with the higher `priority` (`priority` option) or that one that forms bigger chunks.
+
+### Conditions
+
+Modules from the same chunks and cache group will form a new chunk when all conditions are fullfilled.
+
+There are 4 options to configure the conditions:
+
+* `minSize` (default: 30000) Minimum size for a chunk.
+* `minChunks` (default: 1) Minimum number of chunks that share a module before splitting
+* `maxInitialRequests` (default 3) Maximum number of parallel requests at an entrypoint
+* `maxAsyncRequests` (default 5) Maximum number of parallel requests at on-demand loading
+
+### Naming
+
+To control the chunk name of the split chunk the `name` option can be used.
+
+W> When assigning equal names to different split chunks they are merged together. This can be used to put all vendor modules into a single shared chunk, but it's not recommend since it can result in more code downloaded.
+
+The magic value `true` automatically chooses a name based on chunks and cache group key. Elsewise a string or function can be passed.
+
+When the name matches an entrypoint name, the entrypoint is removed.
+
+### Select chunks
+
+With the `chunks` option the selected chunks can be configured. There are 3 values possible `"initial"`, `"async"` and `"all"`. When configured the optimization only selects initial chunks, on-demand chunks or all chunks.
+
+The option `reuseExistingChunk` allows to reuse existing chunks instead of creating a new one when modules match exactly.
+
+This can be controlled per cache group.
+
+### Select modules
+
+The `test` option controls which modules are selected by this cache group. Omitting it selects all modules. It can be a RegExp, string or function.
```

Comment:
It would be nice if this section went more into details about the function version of this option, e.g. what the args to the function are, how to use them, and what the function should return.

Reply:
FWIW, it looks like `test` fn is called with 2 args. `module` and `chunks`. 

https://github.com/webpack/webpack/blob/66ff412f7e2409ee535c2dc79abf4068015b2e22/lib/optimize/SplitChunksPlugin.js#L188

I think `module` is a string that is the actual code in the module. `chunks` seems to be an array  of objects with metadata about each chunk. I'm not sure where this object is documented or defined in code yet.

It seems that `test` functions should return a boolean.

Reply:
Ah found it https://github.com/webpack/webpack/blob/fde018300aa52262c384e937c408d5dd97d62951/lib/Chunk.js#L46

Reply:
Yes, I imagine that boolean decides if the current module goes to a chunk or not.